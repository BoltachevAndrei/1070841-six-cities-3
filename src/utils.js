import {SortTypeList} from './types';

export const capitalizeString = (str) => str[0].toUpperCase() + str.slice(1);

export const extend = (object1, object2) => Object.assign({}, object1, object2);

export const getCities = (data) => Array.from(new Set(data.map((element) => element.city)).values());

export const getOffersByCity = (offers, city) => offers.filter((offer) => offer.city === city);

export const sortOffers = (offers, sortType) => {
  switch (sortType) {
    case SortTypeList.HIGH_TO_LOW:
      return offers.sort((prev, next) => next.price - prev.price);
    case SortTypeList.LOW_TO_HIGH:
      return offers.sort((prev, next) => prev.price - next.price);
    case SortTypeList.TOP_RATED_FIRST:
      return offers.sort((prev, next) => next.rating - prev.rating);
  }
  return offers;
};

export const getCitiesList = (offers) => {
  const citiesNames = Array.from(new Set(offers.map((element) => element.city.name)).values());
  const citiesData = Array.from(offers.values()).map((offer) => offer.city);
  const result = citiesNames.map((name) => citiesData.filter((city) => city.name === name)[0]);
  return result;
};

export const doNothing = () => {
  // do nothing
};
