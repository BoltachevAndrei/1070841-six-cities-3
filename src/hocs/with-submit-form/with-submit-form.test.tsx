import * as React from 'react';
import * as renderer from 'react-test-renderer';
import withSubmitForm from './with-submit-form';
import {doNothing} from '../../utils';

const id = 1;

const isPostingComment = false;

interface MockComponentProps {
  children: React.ReactNode;
}

const MockComponent = (props: MockComponentProps) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

const MockComponentWrapped = withSubmitForm(MockComponent);

it(`withSubmitForm renders correctly`, () => {
  const tree = renderer
    .create((
      <MockComponentWrapped
        id={id}
        isPostingComment={isPostingComment}
        onSubmit={doNothing}
      />
    ), {
      createNodeMock() {
        return {};
      }
    })
    .toJSON();
  expect(tree).toMatchSnapshot();
});
