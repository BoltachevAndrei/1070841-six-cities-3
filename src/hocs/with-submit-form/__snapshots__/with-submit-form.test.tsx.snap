// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withSubmitForm renders correctly 1`] = `<div />`;
